mixer/pkg/protobuf/yaml/resolver.go

```diff
@@ -0,0 +1,59 @@
+// Copyright 2018 Istio Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package yaml
+
+import (
+	"github.com/gogo/protobuf/protoc-gen-gogo/descriptor"
+
+	descriptor2 "istio.io/istio/mixer/pkg/protobuf/descriptor"
+)
+
+// Resolver type is used for finding and resolving references inside proto descriptors.
+type Resolver interface {
```

Comment:
Will help in mocking in case one needs it; that was my intent. However, for my case, I was able to create a testdata file that would return error case from a Resolver. Resolve is intended to be passed around during construction of Byte Encoder or Assembler (under implementation by @mandarjog ) and an interface for it would help for mocking during testing.